### PR TITLE
Add gpu debug builds

### PR DESCRIPTION
**Pull request by pfultz2**
Run a stage using `MIGRAPHX_GPU_DEBUG=1`.

### CODE REVIEW

File: src/targets/gpu/CMakeLists.txt [hunk 1]
```diff
@@ -387,6 +387,13 @@ else()
     message(STATUS "MIOpen does not have find mode api")
 endif()
 
+# set(MIGRAPHX_GPU_DEBUG OFF CACHE BOOL "")
```
Comment (kahmed10) [inline src/targets/gpu/CMakeLists.txt:390]: Should this be uncommented or removed?

File: Jenkinsfile [hunk 2]
```diff
@@ -93,28 +99,32 @@ rocmtest clang_debug: rocmnode('vega') { cmake_build ->
...
         def debug_flags = "-g -O2 -fno-omit-frame-pointer -fsanitize=${sanitizers} -fno-sanitize-recover=${sanitizers}"
-        cmake_build("/opt/rocm/llvm/bin/clang++", "-DCMAKE_BUILD_TYPE=debug -DMIGRAPHX_ENABLE_PYTHON=Off -DMIGRAPHX_ENABLE_GPU=Off -DMIGRAPHX_ENABLE_CPU=On -DCMAKE_CXX_FLAGS_DEBUG='${debug_flags}' -DCMAKE_C_FLAGS_DEBUG='${debug_flags}'")
+        cmake_build(flags: "-DCMAKE_BUILD_TYPE=debug -DMIGRAPHX_ENABLE_PYTHON=Off -DMIGRAPHX_ENABLE_GPU=Off -DMIGRAPHX_ENABLE_CPU=On -DCMAKE_CXX_FLAGS_DEBUG='${debug_flags}' -DCMAKE_C_FLAGS_DEBUG='${debug_flags}'")
     }
 }//, clang_release_navi: rocmnode('navi21') { cmake_build ->
 //    stage('HIP Clang Release Navi') {
-//        cmake_build("/opt/rocm/llvm/bin/clang++", "-DCMAKE_BUILD_TYPE=release")
+//        cmake_build(flags: "-DCMAKE_BUILD_TYPE=release")
 //    }
 //}
```
Comment (turneram) [inline Jenkinsfile:129]: Same question as Khalique had on CMakeLists - is this commented out code meant to be here?

Comment (kahmed10) [reply Jenkinsfile:129]: This is a disabled test that is eventually be re-enabled.

File: src/targets/gpu/CMakeLists.txt (hunk 1 above)
Comment (pfultz2) [reply src/targets/gpu/CMakeLists.txt]: Removed it.

**Review by kahmed10 — APPROVED** [review]

**Review by turneram — APPROVED** [review]

**Pull request merged**